src/realtime/websocket/utils/extract-note-id-from-request-url.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { IncomingMessage } from 'http';
+
+/**
+ * Extracts the note id from the url of the given request.
+ *
+ * @param request The request whose URL should be extracted
+ * @return The extracted note id
+ * @throws Error if the given string isn't a valid realtime URL path
+ */
+export function extractNoteIdFromRequestUrl(request: IncomingMessage): string {
+  if (request.url === undefined) {
+    throw new Error('No URL found in request');
+  }
+  const url = new URL(request.url, 'https://example.org'); //Need a valid origin for the URL constructor to work.
```

Comment:
Improvement suggestion for the comment:

```suggestion
  const url = new URL(request.url, 'https://example.org');
  // A valid domain name is needed for the URL constructor, although not being used here.
  // The example.org domain should be safe to use according to RFC 6761.
```